tutorials/haskell/dotenv-hs/tutorial.md

```diff
@@ -0,0 +1,284 @@
+       expected: [User {id_ = 1, name = "User 1", lastname = "User 1", birthDate = 1990-01-01}
+                 ,User {id_ = 2, name = "User 2", lastname = "User 2", birthDate = 1990-01-01}
+                 ,User {id_ = 3, name = "User 3", lastname = "User 3", birthDate = 1990-01-01}
+                 ,User {id_ = 4, name = "User 4", lastname = "User 4", birthDate = 1990-01-01}]
+
+        but got: [User {id_ = 1, name = "User 1", lastname = "User 1", birthDate = 1990-01-01}
+                 ,User {id_ = 2, name = "User 2", lastname = "User 2", birthDate = 1990-01-01}
+                 ,User {id_ = 3, name = "User 3", lastname = "User 3", birthDate = 1990-01-01}]
+
+Randomized with seed 1355675075
+
+Finished in 0.0494 seconds
+1 example, 1 failure
+
+Completed 2 action(s).
+Test suite failure for package dotenv-hs-example-0.1.0.0
+```
+
+The tests failed, because the database for tests should be used insted of the
+development one. Run the tests again giving a different *dot env* file, in this
+case *.env.test*.
+
+```shell
+$ dotenv "stack test" -e .env.test
+dotenv-hs-example-0.1.0.0: test (suite: dotenv)
+
+
+User.User
+  findAllUsers
+    finds all the users
+
+Finished in 0.0357 seconds
+1 example, 0 failures
+```
+
+Yey! The test pass! The `dotenv` standalone tool can be used with any application
+that needs env vars.
+
+## Dotenv as a Library
+To use `dotenv` as a library import the next modules in the code:
+
+```haskell
+import Configuration.Dotenv
+import Configuration.Dotenv.Types
+```
+
+Define the dotenv configuration, using the `Config` type. For this example:
+
+```haskell
+config :: Config
+config =
+  Config
+    { configExamplePath = ".env.example"
+    , configOverride = True
+    , configPath = ".env"
+    }
+```
+
+Then use the `loadFile` function to load your env vars given `config`.
+
+```haskell
+main :: IO ()
+main = do
+  void $ loadFile config
+  conn <- readConfig >>= getConnection
+  users <- findAllUsers conn
+  forM_ users print
+```
+
+As simple as that! Let's execute the application.
+
+```shell
+$ stack exec dotenv-code
+User {id_ = 1, name = "User 1", lastname = "User 1", birthDate = 1990-01-01}
+User {id_ = 2, name = "User 2", lastname = "User 2", birthDate = 1990-01-01}
+User {id_ = 3, name = "User 3", lastname = "User 3", birthDate = 1990-01-01}
+```
+
+The env vars will be in the environment before the rest of the application starts.
+
+# Conclusion
+Safety and complete set up configuration before execution is assured with dotenv
+Safey, because the execution of the application will stop if the necesary env
+vars are not defined the environment or in the dotenv file. And complete
+configuration, because the necesary env vars will be set up after dotenv execution.
+Now, go ahead and use `dotenv` everywhere!
+
+[dotenv]: https://github.com/stackbuilders/dotenv-hs
+[example-code]: https://github.com/CristhianMotoche/tutorials/tree/how_to_use_dotenv/tutorials/haskell/dotenv-hs/code
```

Comment:
This code should live under the **stackbuilders** organization. There is no benefit in adding an external reference here.